docs/explanation/parts.rst

```diff
@@ -0,0 +1,263 @@
+In YAML format, this can take either of the following forms:
+
+.. code:: yaml
+
+   build-environment: [MESSAGE: "Hello world", NAME: "Craft Parts"]
+
+   build-environment:
+     - MESSAGE: "Hello world"
+     - NAME: "Craft Parts"
+
+The ``build-attributes`` key allows a number of standard customisations to be
+applied to the build. Some of these are used to address issues that occur in
+specific situations; others, such as ``debug`` are generally useful.
+
+By default, Craft Parts builds independent parts in parallel. This can be
+disabled by setting the ``disable-parallel`` key to ``True``.
+
+The ``override-build`` key is used to override the build process provided by
+a plugin, and it can be used to replace or extend it.
+
+Build products
+~~~~~~~~~~~~~~
+
+After a part has been built in the *build* step, the build products are
+prepared in three further steps.
+
+If the ``organize`` key is specified, it is used to move files
+specify a list of mappings between files
+in the part key-value pairs
+
+   * How to manage the different steps in the build process
+
+    override-stage string
+    override-prime string
+
+
+
+   * Filesets
+
+ * How artifacts are exported/installed
+
+
+
+
+    stage-snaps ref: #/definitions/grammar-array
+    stage-packages ref: #/definitions/grammar-array
+    organize
+    filesets
+    stage array with at least 1 unique item(s) of string
+    prime array with at least 1 unique item(s) of string
+
+..    parse-info array with at least 1 unique item(s) of string
+
+.. _build-order:
+
+Defining the build order
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, when more than one part is specified, they are built in alphabetical order unless there are dependencies between parts.
+
+One way to define a dependency for a part is to use the ``after`` key in
+a part's definition to specify a list of parts that it will be built after. The parts whose names are supplied in the list will be *built and staged*
+before the part is built.
+
+This is covered in detail in :ref:`part_processing_order`.
+
+.. Overriding aspects of the build
+.. ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+How parts are built
+-------------------
+
+As described in :ref:`lifecycle`, parts are built in a sequence of steps:
+*pull*, *overlay*, *build*, *organize*, *stage* and *prime*.
+
+A part is built in a clean environment to ensure that only the base and its
+dependencies are present, avoiding contamination from partial builds and
+side effects from other builds. The environment is a file system in a
+container where the root user's home directory is populated with a number of
+subdirectories and configured to use snaps.
+
+.. ### Verify that snap is available in general for non-Snapcraft builds.
+
+Initially, before the *pull* step is run, the ``/root`` directory contains a
+``project`` directory containing the files for the project to be built.
+
+The pull step
+~~~~~~~~~~~~~
+
+When the *pull* step is run the :doc:`sources <Source>` are obtained using
```

Comment:
This is a bit confusing. I would use a chronological bullet list to explain what happens.